### ISSUE TITLE
[Question] Transfer learning with trained Resnet34

### ISSUE THREAD

**Issue by zul96a** [leondgarse/Keras_insightface#49]
Hi again! 

So far I have been following your method in training using resnet34 (https://github.com/leondgarse/Keras_insightface/discussions/36), and it works well as I anticipated. 

I'm just wondering what are the steps that I can do if I want to do fine tuning with [this](https://github.com/leondgarse/Keras_insightface/releases/download/v1.0.0/resnet34_MXNET_E_SGD_REG_1e3_on_batch_true_lr1e1_random0_arc_S32_E1_BS512_casia_basic_agedb_30_epoch_36_0.949500.h5) model that you posted on the main page. My strategy is to use the existing model here and fine tune it with my set of images , 10k in total of around 4k persons, instead of training from scratch.

As I understand, the model that I linked is already trained with CASIA webface dataset and has achieved a very high accuracy on lfw, cfp_fp, and agedb_30. I wonder if it's possible to fine tune it to incorporate with my dataset. Thanks!

**Comment by leondgarse** [comment]
Hi, just back from our vacation, don't know how's your fine-tuning going on.
Basically, it's like:
- Train some epochs with freezed early layers in basic_model. Then train with all unfreezed.
- Train model using a smaller learning_rate.
```py
import losses, train, models
# keras.mixed_precision.set_global_policy("mixed_float16")  # May try if float16 can speed up

data_path ='/your_own_dataset_path'
eval_paths = ['your_own_eval_paths_if_any']

basic_model = keras.models.load_model("r34_pre_trained_model.h5")  # load the basic_model, inputs --> embeddings

# Freeze layers before "E" / "GDC" / "GAP" head layers.
# May change the the number `-6` on own needs. See `basic_model.summary()` for layers detail.
# May also try freeze the whole basic_model by `basic_model.trainable = False` first.
for ii in basic_model.layers[:-6]:
    ii.trainable = False

# Set a lower learning_rate, like 1e-3
tt = train.Train(data_path, eval_paths=eval_paths, save_path='new_save_path.h5',
    basic_model=basic_model, model=None, lr_base=1e-3, lr_decay=0.5, lr_decay_steps=16, lr_min=1e-5,
    batch_size=1024, random_status=0, eval_freq=2000, output_weight_decay=1)

optimizer = keras.optimizers.SGD(learning_rate=0.1, momentum=0.9)
tt.train({"loss": losses.ArcfaceLoss(scale=64), "epoch": 17, "optimizer": optimizer}, 0)
```
Those hyper-parameters may depends on your dataset and result. I'm not sure with all of them. You may also refer to [finetune issues from deepinsight/insightface](https://github.com/deepinsight/insightface/issues?q=finetune).

**Comment by leondgarse** [comment]
Ya, I just copy those from other issues. :)
I think 3.22 is pretty normal, like the r34 model I shared is 3.7672, the brown line [here the r34 training script](https://github.com/leondgarse/Keras_insightface/discussions/36). It depends on your dropout / random status / weight decay and if l2 loss is added.

**Comment by leondgarse** [comment]
Ya, it's cosine distance been used.
